nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ParameterContextResource.java

```diff
@@ -867,6 +867,12 @@ private ParameterContextEntity updateParameterContext(final AsynchronousWebReque
             updatedEntity = performParameterContextUpdate(asyncRequest, uri, replicateRequest, revision, updatedContextEntity);
             logger.info("Successfully updated Parameter Context with ID {}", updatedContextEntity.getId());
         } finally {
+            if (!asyncRequest.isCancelled()) {
```

Comment:
@rfellows I am not quite sure as I'm not familiar enough with this particular set of code but if this is attempting a synchronous validation operation so that when we attempt restart we should be good - this might not be the way to go as we tried to move away from such things.  Rather, we'd want to have restart happen after validation has had some time to occur and/or we attempt it a few times to allow that to happen (before we give up).  If someone else can review/comment that would be good too if there is greater familiarity

Reply:
@joewitt We do certainly want to avoid making synchronous requests from a web thread because we don't want to block the response from going back to the user. In this particular case, this code is executed from a separate background thread, so blocking here should be okay.
That being said, @rfellows this will perform validation only on the cluster coordinator, where this will occur. We need to ensure that in a cluster environment that we wait for validation to complete across all nodes in the cluster. So we need to implement this from within the two implementations of the `ComponentLifecycle` interfaces. `LocalComponentLifecycle` can implement essentially the same way, calling `serviceFacade#validateComponents` but `ClusterReplicationComponentLifecycle` will need to basically just poll, waiting for the validation to complete, similar to how it has the `waitForProcessorStatus` method waits for the processors to be started or stopped.